content/300-guides/300-upgrade-guides/200-upgrading-versions/600-upgrading-to-prisma-5.mdx

```diff
@@ -661,6 +661,23 @@ process.on('SIGTERM', exitHandler)
 process.on('SIGUSR2', exitHandler)
 ```
 
+If you're using the `beforeExit` hook in NestJs, you can upgrade to Prisma 5 by removing the `enableShutdownHooks` in your application:
+
+```diff file="prisma.service.ts"
+@Injectable()
+export class PrismaService extends PrismaClient implements OnModuleInit {
+  async onModuleInit() {
+    await this.$connect()
+  }
+
+-  async enableShutdownHooks(app: INestApplication) {
+-    this.$on('beforeExit', async () => {
+-      await app.close()
+-    })
+-  }
+}
+```
```

Comment:
````suggestion
```

Instead, use the built-in `enableShutdownHooks` method in NestJS if you need to handle lifecycle events:

```diff file="main.ts"
- prismaService.enableShutdownHooks(app)
+ app.enableShutdownHooks()
```
````